### PR TITLE
Nested errors

### PR DESCRIPTION
**Pull request by jondeandres**
The Rollbar API support sending to it nested exceptions, so you can have an error `NetworkTimeout` but the reported error is `PurchaseFailed` both will appear in Rollbar as nested. You can then known that a `PurchaseFailed` error occurred and the cause of it.

Because JavaScript doesn't support built-in nested errors, we provide a `rollbar.Error` class that you can inherit from, so you can send to the Rollbar API nested exceptions. Example:

```javascript
var rollbar = require('.rollbar');
var util = require('util');


function NetworkTimeout(message, nested) {
  rollbar.Error.call(this, message, nested);
}

util.inherits(NetworkTimeout, rollbar.Error);


function PurchaseFailed(message, nested) {
  rollbar.Error.call(this, message, nested);
}

util.inherits(PurchaseFailed, rollbar.Error);


function sendPurchase(data, cb) {
  var err = new NetworkTimeout('Error sending data to payment gateway');

  cb(err, null);
}


function doPurchase(data) {
  sendPurchase(data, function(err, res) {
    if (err) {
      rollbar.handleError(new PurchaseFailed('Purchase has failed', err));
    }
  });
}

doPurchase({
  id: 10,
  user_id: 1,
  amount: 5
});
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,6 +260,54 @@ Default: `true`
 To show operational messages, include `Rollbar:*` (or, for only some messages, `Rollbar:log` or `Rollbar:error`) in your `DEBUG` environment variable.
 
 
+### Nested exceptions
+
+The Rollbar API support sending to it nested exceptions, so you can have an error `NetworkTimeout` but the reported error is `PurchaseFailed` both will appear in Rollbar as nested. You can then known that a `PurchaseFailed` error occurred and the cause of it.
```
Comment (coryvirok) [inline README.md]: Reworded:

```
The Rollbar API supports nested exceptions. This allows you to report an error along with the original cause as a nested exception.
```

File: README.md [hunk 2]
```diff
@@ -260,6 +260,54 @@ Default: `true`
...
+
+Because JavaScript doesn't support built-in nested errors, we provide a `rollbar.Error` class that you can inherit from, so you can send to the Rollbar API nested exceptions. Here an example:
```
Comment (coryvirok) [inline README.md]: Reworded:

> In order to create a nested error you should use the `rollbar.Error` class provided by this library.
> E.g.
>```js
>// Code from below
>```

File: README.md (hunk 1 above)
Comment (jondeandres) [inline README.md]: Changed

File: README.md (hunk 2 above)
Comment (jondeandres) [inline README.md]: Changed

**Pull request merged**